api/features/feature_health/services.py

```
@@ -0,0 +1,88 @@
+import typing
+
+import structlog
+
+from environments.models import Environment
+from features.feature_health.constants import UNHEALTHY_TAG_COLOUR
+from features.feature_health.models import (
+    FeatureHealthEvent,
+    FeatureHealthEventType,
+    FeatureHealthProvider,
+    FeatureHealthProviderName,
+)
+from features.feature_health.providers import sample
+from features.models import Feature
+from projects.tags.models import Tag, TagType
+
+if typing.TYPE_CHECKING:
+    from features.feature_health.types import FeatureHealthProviderResponse
+
+logger = structlog.get_logger("feature_health")
+
+
+def get_provider_response(
+    provider: FeatureHealthProvider, payload: str
+) -> "FeatureHealthProviderResponse | None":
+    if provider.name == FeatureHealthProviderName.SAMPLE.value:
+        return sample.map_payload_to_provider_response(payload)
+    logger.error(
+        "invalid-feature-health-provider-requested",
+        provider_name=provider.name,
+        provider_id=provider.uuid,
+    )
+    return None
+
+
+def create_feature_health_event_from_provider(
+    provider: FeatureHealthProvider,
+    payload: str,
+) -> FeatureHealthEvent | None:
+    try:
+        response = get_provider_response(provider, payload)
+    except Exception as exc:
```

Why do we need to catch all exceptions here?

---

Good point, clarified. Also changed the event name a bit to better reflect the expected mapping errors.